agent/src/main.cpp

```diff
@@ -52,9 +53,51 @@ int signalHandler(void *data) {
     return 0;
 }
 
+void cleanGhostContainers(void)
+{
+    char **containers = nullptr;
+    struct lxc_container **cret = nullptr;
+    const char * basePath = lxc_get_global_config_item("lxc.lxcpath");
+    auto num = list_all_containers(basePath, &containers, &cret);
+    if (num > 0 && num < std::numeric_limits<int>::max()) {
+        log_warning() << num << " ghost containers found";
+        for (; num; num--) {
+            log_debug() << "Ghost container named " << containers[num-1] << " will be deleted";
+
+            const char * state = cret[num-1]->state(cret[num-1]);
+            if ("RUNNING" == std::string(state)) {
+                bool success = cret[num-1]->stop(cret[num-1]);
```

Comment:
Dunno if it would make more sense to use shutdown() before calling stop(). What do you think?